### PR TITLE
RSDK-1902 Random panic occurring somewhere in CBiRRT method

### PR DESCRIPTION
**Pull request by biotinker**
Update NearestNeighbor to run on multiple maps in parallel, and properly handle the return of a `nil` from NN which can only be due to the ctx being cancelled

### CODE REVIEW

File: motionplan/cBiRRT.go [hunk 1]
```diff
@@ -198,10 +198,23 @@ func (mp *cBiRRTMotionPlanner) rrtBackgroundRunner(
 		default:
 		}
 
-		tryExtend := func(target []referenceframe.Input) (node, node, float64) {
+		tryExtend := func(target []referenceframe.Input) (node, node, float64, error) {
```
Comment (raybjork) [inline motionplan/cBiRRT.go:201]: Returning 4 values here seems less than ideal.  I'm wondering if we need to return `reachedDelta` here.  Can we instead just return the nodes and if the algorithm needs to compute this after the fact then it always can.

**Review by raybjork — COMMENTED** [review]
Looks good.  Just left one nit.  

Do we not need to change anything for RRT* for this?

File: motionplan/cBiRRT.go (hunk 1 above)
Comment (biotinker) [reply motionplan/cBiRRT.go:201]: You're right; we don't need to be computing the delta in here. Will update.

**Review by raybjork — APPROVED** [review]

**Review by spieswl — APPROVED** [review]
LGTM

**Review by nfranczak — APPROVED** [review]

**Pull request merged**